checker/jdk/index/src/java/util/Collections.java

```diff
@@ -3349,21 +3349,26 @@ public void remove() {
      * @return an immutable list containing only the specified object.
      * @since 1.3
      */
-    public static <T> @MinLen(1) List<T> singletonList(T o) {
+    public static <T> @ArrayLen(1) List<T> singletonList(T o) {
         return new SingletonList<>(o);
     }
 
     /**
      * @serial include
      */
-    private static class SingletonList<E>
+    private static @ArrayLen(1) class SingletonList<E>
         extends AbstractList<E>
         implements RandomAccess, Serializable {
 
         private static final long serialVersionUID = 3093736618740652951L;
 
         private final E element;
 
+        @SuppressWarnings({"inconsistent.constructor.type", "super.invocation.invalid"}) /*
```

Comment:
To indicate which comment is about which warning, please format the comments like this:

```
        @SuppressWarnings({
	 "inconsistent.constructor.type", // explanation about inconsistent.constructor.type 
         "super.invocation.invalid" // explanation about super.invocation.invalid
        })
```

Reply:
I have made this change now